pkg/asset/installconfig/nutanix/validation.go

```diff
@@ -0,0 +1,63 @@
+package nutanix
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types"
+	nutanixtypes "github.com/openshift/installer/pkg/types/nutanix"
+	"github.com/openshift/installer/pkg/types/nutanix/validation"
+)
+
+// Validate executes platform-specific validation.
+func Validate(ic *types.InstallConfig) error {
+	if ic.Platform.Nutanix == nil {
+		return errors.New(field.Required(field.NewPath("platform", "nutanix"), "nutanix validation requires a nutanix platform configuration").Error())
+	}
+
+	p := ic.Platform.Nutanix
+	if errs := validation.ValidatePlatform(p, field.NewPath("platform").Child("nutanix")); len(errs) != 0 {
+		return errs.ToAggregate()
+	}
+
+	_, err := nutanixtypes.CreateNutanixClient(context.TODO(), p.PrismCentral, p.Port, p.Username, p.Password)
+	if err != nil {
+		return errors.New(field.InternalError(field.NewPath("platform", "nutanix"), errors.Wrapf(err, "unable to connect to Prism Central %s.", p.PrismCentral)).Error())
+	}
+
+	return nil
+}
+
+// ValidateForProvisioning performs platform validation specifically for installer-
+// provisioned infrastructure. In this case, self-hosted networking is a requirement
+// when the installer creates infrastructure for nutanix clusters.
+func ValidateForProvisioning(ic *types.InstallConfig) error {
+	if ic.Platform.Nutanix == nil {
+		return errors.New(field.Required(field.NewPath("platform", "nutanix"), "nutanix validation requires a nutanix platform configuration").Error())
+	}
+
+	p := ic.Platform.Nutanix
+	nc, err := nutanixtypes.CreateNutanixClient(context.TODO(),
+		p.PrismCentral,
+		p.Port,
+		p.Username,
+		p.Password)
+	if err != nil {
+		return errors.New(field.InternalError(field.NewPath("platform", "nutanix"), errors.Wrapf(err, "unable to connect to Prism Central %s.", p.PrismCentral)).Error())
+	}
+
+	_, err = nc.V3.GetSubnet(p.SubnetUUID)
+	if err != nil {
+		return errors.New(fmt.Sprintf("subnet UUID %s does not correspond to a valid subnet in Prism", p.SubnetUUID))
+	}
+
+	_, err = nc.V3.GetCluster(p.PrismElementUUID)
+	if err != nil {
+		return errors.New(fmt.Sprintf("prism element UUID %s does not correspond to a valid subnet in Prism", p.PrismElementUUID))
```

Comment:
"valid subnet"? Is that a copy error?
```suggestion
		return errors.New(fmt.Sprintf("prism element UUID %s does not correspond to a valid subnet in Prism", p.PrismElementUUID))
```

Reply:
seems so; good catch.